### PR TITLE
Nodes layer

### PR DESCRIPTION
**Pull request by pshemass**
- I converted `Nodes` to layer
- introduced `ConversationId` that generates id for each message 
- organized messages and added test for serialization
- tests added for FailureDetection

### CODE REVIEW

File: keeper/src/main/scala/zio/keeper/errors.scala [hunk 1]
```diff
@@ -4,7 +4,7 @@ import zio.duration.Duration
 import zio.keeper.transport.Address
 import zio.nio.core.SocketAddress
 
-sealed abstract class Error(val msg: String = "") {
+sealed abstract class Error(val msg: String = "") extends Throwable {
```
Comment (mijicd) [inline keeper/src/main/scala/zio/keeper/errors.scala:7]: I'd extend `Exception` (or `RuntimeException`). Not sure whether or not it make sense to disable stack trace accumulation, e.g. `extends Exception(msg, null, false, false)`.

File: keeper/src/main/scala/zio/keeper/membership/swim/ConversationId.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package zio.keeper.membership.swim
+
+import zio.UIO
+import zio.ZLayer
+import zio.Ref
+import zio.ULayer
+import zio.ZIO
+import zio.URIO
```
Comment (mijicd) [inline keeper/src/main/scala/zio/keeper/membership/swim/ConversationId.scala:8]:
```suggestion
import zio._
```

File: keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala [hunk 3]
```diff
@@ -5,121 +5,89 @@ import zio.clock.Clock
...
+     * Node state for given NodeId.
+     */
+    def nodeState(id: NodeAddress): IO[Error, NodeState]
+
+    def numberOfNodes: UIO[Int]
+
+    /**
+     * Lists members that are in healthy state.
+     */
+    def healthyNodes: UIO[List[(NodeAddress, NodeState)]]
```
Comment (mijicd) [inline keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala:60]: Does it make sense to change to `Chunk`? Long-term goal in core is to migrate wherever possible. No need to do this now, just asking.

File: keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala [hunk 4]
```diff
@@ -5,121 +5,89 @@ import zio.clock.Clock
...
+     * Returns next Healthy node.
+     */
+    val next: UIO[Option[NodeAddress]]
+
+    /**
+     * Node state for given NodeId.
+     */
+    def nodeState(id: NodeAddress): IO[Error, NodeState]
+
+    def numberOfNodes: UIO[Int]
```
Comment (mijicd) [inline keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala:55]:
```suggestion
    val numberOfNodes: UIO[Int]
```

File: keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala (hunk 3 above)
Comment (pshemass) [reply keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala:60]: do you mean migrate ```List``` to ```Chunk```?

File: keeper/src/main/scala/zio/keeper/errors.scala (hunk 1 above)
Comment (pshemass) [reply keeper/src/main/scala/zio/keeper/errors.scala:7]: @mschuwalow what do you think ?

Comment (mschuwalow) [reply keeper/src/main/scala/zio/keeper/errors.scala:7]: I'm personally a fan of keeping stack traces. Main reason for dropping them would be performance, no? In that case let's keep them for now and rip them out when we do a performance pass

File: keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala (hunk 3 above)
Comment (mschuwalow) [reply keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala:60]: Yeah, I think we should start defaulting to Chunk.

File: keeper/src/test/scala/zio/keeper/membership/SwimSpec.scala [hunk 5]
```diff
@@ -122,7 +123,7 @@ object SwimSpec {
               )
             )
           )
-      }.provideLayer(Clock.live ++ logging ++ (logging >>> TestDiscovery.live))
-    )
+      }.provideLayer(Clock.live ++ logging ++ (logging ++ Random.live >>> TestDiscovery.live))
+    ) @@ TestAspect.ignore
```
Comment (mschuwalow) [inline keeper/src/test/scala/zio/keeper/membership/SwimSpec.scala:127]: What happened here?

Comment (pshemass) [reply keeper/src/test/scala/zio/keeper/membership/SwimSpec.scala:127]: this is the test that is hanging on CI. locally I can run it with `nonFlaky` aspect but on CI it hangs :(

File: keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala (hunk 3 above)
Comment (pshemass) [reply keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala:60]: let me try it.

File: keeper/src/main/scala/zio/keeper/errors.scala (hunk 1 above)
Comment (pshemass) [reply keeper/src/main/scala/zio/keeper/errors.scala:7]: so I will change that to `Exception` and keep stacktrace for the time being.

Comment (pshemass) [reply keeper/src/main/scala/zio/keeper/errors.scala]: done

File: keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala (hunk 3 above)
Comment (pshemass) [reply keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala:60]: let do this when `TMap` will have `toChunk` method.

Comment (mijicd) [reply keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala:60]: It's already there as of RC19.

Comment (pshemass) [reply keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala:60]: let do this in new PR

**Pull request merged**